docs/examples/POWGEN_data_reduction.ipynb

```diff
@@ -0,0 +1,424 @@
+   "source": [
+    "pipeline[OutFilename] = 'reduced.xye'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1eb08353-2bbd-4692-9f8d-7349ddc5d400",
+   "metadata": {},
+   "source": [
+    "And use the pipeline to write the file.\n",
+    "Note that this recomputes the result!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d35257da-b7a8-47e3-bd90-3bdf98ee539b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pipeline.bind_and_call(save_xye)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0420ed62-aa60-466f-a823-f608ddc3abc5",
+   "metadata": {},
+   "source": [
+    "### Compute intermediate results\n",
+    "\n",
+    "For inspection and debugging purposes, we can also compute intermediate results.\n",
+    "To avoid repeated computation (including costly loading of files), we can request multiple results at once, including the final result, if desired.\n",
+    "For example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "300d11ce-203b-4c65-a7f6-9df5132f9830",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "results = pipeline.compute((\n",
+    "    RawData[SampleRun],\n",
+    "    FilteredData[SampleRun],\n",
+    "    FilteredData[VanadiumRun],\n",
+    "))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b57d17b6-926a-4b75-8221-5ed27f59997d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "results[RawData[SampleRun]]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "57d72dd9-7f21-4691-b478-109194cc9c13",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "scn.instrument_view(results[RawData[SampleRun]].hist())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e5547f3e-908c-4dcb-a700-c4aa5098073c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tof_data = sc.DataGroup(\n",
+    "    sample=results[RawData[SampleRun]].bins.concat('spectrum'),\n",
+    "    vanadium=results[FilteredData[VanadiumRun]].bins.concat('spectrum'),\n",
+    ")\n",
+    "tof_data.hist(tof=100).plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4048b5ad-acfa-4584-8cf6-8ebe2cba6801",
+   "metadata": {},
+   "source": [
+    "## Group by scattering angle\n",
+    "\n",
+    "The above pipeline merges all instrument pixels to produce a 1d d-spacing curve.\n",
+    "If instead we want to group into $2\\theta$ bins, we can alter the pipeline by replacing the grouping step:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d24c8bf8-ad59-4211-ae6a-3ee29b0556a3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ess.diffraction.grouping import group_by_two_theta, merge_all_pixels\n",
+    "\n",
+    "grouping_providers = list(providers)\n",
+    "grouping_providers.remove(merge_all_pixels)\n",
+    "grouping_providers = (*grouping_providers, group_by_two_theta)"
```

Comment:
Would it make more sense to have `TwoThetaBins` as an optional input to a single provider, which falls back to merging all pixels of no bins provided? This would avoid the need to modify the provider list.

Reply:
Yes, swapping a provider does not fix the problem, so a different type would work.

Reply:
According to my discussion with Andrew about CIF, it seems that we will have to focus in some way that still needs to be determined. It sounds like grouping all pixels is quite useless. But using 2theta is likely also not a good approach, at least for DREAM.

So I wouldn't put too much thought into finding a good solution now. Since this is not the last step in the pipeline, I'd rather keep the unified output type until we know more.